flask_hw5/blueprints/rooms/rooms_routes.py

```diff
@@ -0,0 +1,57 @@
+from flask_restful import Resource
+from sqlalchemy.exc import SQLAlchemyError
+
+from blueprints.rooms.room_arg_parser import (room_parser,
+                                              room_parser_post,
+                                              room_parser_patch)
+from blueprints.rooms.room_object_serlzer import marshal_with_decor
+from blueprints.rooms.rooms_model import Room
+from hotel_db import db
+
+
+class Rooms(Resource):
+    @marshal_with_decor
+    def get(self):
+        room_filters = room_parser.parse_args()
+        if any(tuple(room_filters.values())):
+            return Room.query.filter_by(**{k: v
+                                           for k, v in room_filters.items()
+                                           if v}).first_or_404()
+        return Room.query.all(), 200
+
+    @marshal_with_decor
+    def post(self):
+        new_room_values = room_parser_post.parse_args()
+        room_to_add = Room(**new_room_values)
+        db.session.add(room_to_add)
+        try:
+            db.session.commit()
+        except SQLAlchemyError:
+            return 'Something wrong happened on our side. Please, try later!'
+        return room_to_add, 200
+
+    def patch(self, number):
+        room_info_to_update = room_parser_patch.parse_args()
+        room_to_update = Room.query.filter_by(number=number).first_or_404()
+        if not room_to_update:
+            return 'Sorry the room you are looking for was not found.'
```

Comment:
code 400 or 404

Reply:
Updated. if else in this case is not required. first_or_404 method manages this problem.